sycl/test/basic_tests/stream.cpp

```diff
@@ -208,7 +208,7 @@ int main() {
     Queue.submit([&](handler &CGH) {
       stream Out(1024, 80, CGH);
       CGH.parallel_for<class stream_string>(
-          range<1>(10), [=](id<1> i) { Out << "Hello, World!\n"; });
+          range<1>(10), [=](id<1> i) { Out << "Hello, World!" << endl; });
```

Comment:
What is the difference here?

Reply:
std::endl usually flushes after adding a newline, this commit looks like it adds a fall-through to a flush for the SYCL implementation of endl to make it equivalent. So it's making sure it flushes and adds a newline now if I understand the change to the << operator correctly!

Reply:
Thanks. So what this "stream endl" implementation actually does to flush? Does it write data to the global buffer and changes the atomic offset?
What happens if I don't use endl at all?

Reply:
Ok, I've found the code that implements this, and it seems it indeed writes data to global buffer.
Still wonder what happens if we don't use endl.

Reply:
If endl is not used then data is flushed to global buffer in the destructor of the stream object.

Reply:
The reason I ask is because you've replaced all "\n" with endl. Can you keep "\n" at least in some cases to test it?

Reply:
I have added separate test for this. Problem is that commands cleanup is not implemented in scheduler yet (there is a TODO),  this is the reason why stream is not flushed on host device for this test case, I have disabled this test for host and added todo.

Reply:
Yes, you are right, for the host device output that is not followed by endl gets lost.

Reply:
Described this in commit message and added note to the header.